### PR TITLE
MOB-3144 - Display cancellation fee on Trip and Ride Details screens

### PR DESCRIPTION
**Pull request by jsm20**
## JIRA
[MOB-3144](https://karhoo.atlassian.net/browse/MOB-3144)

## DESCRIPTION
Display the cancellation fee to the user and the option to accept or decline the cancellation.

## Types of changes

What types of changes does your code introduce to the project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix 
- [x] New feature 
- [ ] Tech Debt
- [ ] Breaking change 
- [ ] Documentation Update
- [ ] Other: *define what type of change it is*

## CHECKLIST

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. This is simply a reminder of what we are going to look for before merging code._
Please review the guidelines for contributing to this repository.

- [x] All work relating to the ticket is complete (Code complete/Acceptance Criteria met)
- [x] Lint and unit tests pass locally (if appropriate)
- [ ] Added tests that prove the fix is effective or that feature works (if appropriate)
- [ ] Documentation (if appropriate)

### CODE REVIEW

File: KarhooUISDK/Classes/CancelRideBehaviour.swift [hunk 1]
```diff
@@ -12,51 +12,88 @@ import KarhooSDK
...
+                }
+            })
     }
 
     private func callFleetPressed() {
         phoneNumberCaller.call(number: trip.fleetInfo.phoneNumber)
     }
 
-    private func showConfirmCancelRideAlert() {
+    private func showConfirmCancelRideAlert(callback: @escaping CallbackClosure<KarhooVoid>) {
```
Comment (mohadian) [inline KarhooUISDK/Classes/CancelRideBehaviour.swift:96]: Do we need the callback?

File: KarhooUISDK/Screens/Rides/RidesList/RideDetails/KarhooRideDetailsPresenter.swift [hunk 2]
```diff
@@ -22,6 +22,7 @@ final class KarhooRideDetailsPresenter: RideDetailsPresenter {
     private let analyticsService: AnalyticsService
     private let tripFeedbackScreenBuilder: TripFeedbackScreenBuilder
     private let tripRatingCache: TripRatingCache
+    private var alertHandler: AlertHandlerProtocol?
```
Comment (mohadian) [inline KarhooUISDK/Screens/Rides/RidesList/RideDetails/KarhooRideDetailsPresenter.swift:25]: Do we need the alertHandler ? does not seem to be used in this class

File: KarhooUISDK/Classes/CancelRideBehaviour.swift (hunk 1 above)
Comment (jsm20) [reply KarhooUISDK/Classes/CancelRideBehaviour.swift]: In fact, the whole method isn't required so deleted.

File: KarhooUISDK/Screens/Rides/RidesList/RideDetails/KarhooRideDetailsPresenter.swift (hunk 2 above)
Comment (jsm20) [reply KarhooUISDK/Screens/Rides/RidesList/RideDetails/KarhooRideDetailsPresenter.swift]: Nope. Deleted.

**Review by mohadian — APPROVED** [review]

**Pull request merged**